src/services/pocket-relayer.ts

```diff
@@ -637,12 +656,80 @@ export class PocketRelayer {
         blockchainSyncAllowance,
         blockchainIDCheck,
         blockchainID,
+        blockchainLogLimitBlocks,
       })
     } else {
       throw new HttpErrors.BadRequest('Incorrect blockchain: ' + this.host)
     }
   }
 
+  async enforceLimits(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    parsedRawData: Record<string, any>,
+    blockchain: string,
+    logLimitBlocks: number
+  ): Promise<string | Error> {
+    if (parsedRawData.method === 'eth_getLogs') {
+      let toBlock: number
+      let fromBlock: number
+      let isToBlockHex = false
+      let isFromBlockHex = false
+      const altruistUrl = String(this.altruists[blockchain])
+      const [{ fromBlock: fromBlockParam, toBlock: toBlockParam }] = parsedRawData.params as [
+        { fromBlock: string; toBlock: string }
+      ]
+
+      if (toBlockParam !== undefined && toBlockParam !== 'latest') {
+        toBlock = blockHexToDecimal(toBlockParam)
+        isToBlockHex = true
+      }
+      if (fromBlockParam !== undefined && fromBlockParam !== 'latest') {
+        fromBlock = blockHexToDecimal(fromBlockParam)
+        isFromBlockHex = true
+      }
+
+      if (altruistUrl !== 'undefined') {
```

Comment:
Can we only make this check if necessary?  Only if 'latest' is used as a variable. It seems we don't need this call if we know the block numbers already.